core/src/main/scala/org/apache/spark/h2o/utils/H2OContextRestAPIUtils.scala

```diff
@@ -279,7 +279,12 @@ trait H2OContextRestAPIUtils extends H2OContextUtils {
         case unknown => throw new IllegalArgumentException(s"Unsupported HTTP request type $unknown")
       }
       getCredentials(conf).foreach(creds => request.setHeader(HttpHeaders.AUTHORIZATION, creds))
-      httpClient.execute(request).getEntity.getContent
+      val result = httpClient.execute(request)
+      val statusCode = result.getStatusLine.getStatusCode
+      statusCode match {
+        case 401 => throw new Exception("Unauthorized")
```

Comment:
Unimportant: WDYT about introducing a new exception type for REST-based errors?

Reply:
yup, like it, will do as part of this change, thanks!